### PR TITLE
8258972: unexpected compilation error with generic sealed interface

### PR DESCRIPTION
**Pull request by vicente-romero-oracle**
Hi,

Please review this fix to the implementation of section `5.1.6.1 Allowed Narrowing Reference Conversion` of the Sealed Classes spec [1](http://cr.openjdk.java.net/~gbierman/jep397/jep397-20201204/specs/sealed-classes-jls.html#jls-5.1.6.1). IMO the compiler was not implementing the spec to the letter in particular the inheritance relation between a sealed class and a subclass should be tested ignoring type arguments which is what this patch is doing. Comments?

Thanks,
Vicente

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8258972](https://bugs.openjdk.java.net/browse/JDK-8258972): unexpected compilation error with generic sealed interface

### Reviewers
 * [Jan Lahoda](https://openjdk.java.net/census#jlahoda) (@lahodaj - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk16 pull/72/head:pull/72`
`$ git checkout pull/72`

### CODE REVIEW

File: src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java [hunk 1]
```diff
@@ -1667,7 +1667,7 @@ public boolean isCastable(Type t, Type s, Warner warn) {
     }
     // where
         private boolean areDisjoint(ClassSymbol ts, ClassSymbol ss) {
-            if (isSubtype(ts.type, ss.type)) {
+            if (isSubtype(erasure(ts.type), erasure(ss.type))) {
```
Comment (lahodaj) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java:1670]: There is another `isSubtype` below - could you please check if it needs the same erasure handling?

Comment (vicente-romero-oracle) [reply src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java:1670]: I agree, done

**Review by lahodaj — APPROVED** [review]
Looks good to me.

**Pull request closed without merging**